staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -204,6 +213,10 @@ func (f *fancyResponseWriterDelegator) Flush() {
 func (f *fancyResponseWriterDelegator) Hijack() (net.Conn, *bufio.ReadWriter, error) {
 	// fake a response status before protocol switch happens
 	f.processCode(http.StatusSwitchingProtocols)
+
+	// TODO(caoshufeng): we can't call WriteHeader before Hijack(), because WriteHeader may be called later:
```

Comment:
@sttts What do you think about that? Should it be a TODO or is it working as intended and we shouldn't send audit-id for hijack'd requests?